lib/elixir/lib/application.ex

```diff
@@ -123,8 +123,12 @@ defmodule Application do
   @doc """
   Returns the value for `key` in `app`'s environment.
 
-  If the specified application is not loaded, or the configuration parameter
-  does not exist, the function returns the `default` value.
+  If the configuration parameter does not exist, the function returns the
+  `default` value.
+
+  WARNING: Some tools/environments load only the configuration of loaded
+  applications, so you should only pass the name of loaded applications
+  as `app` or you risk getting differing behavior in different environments.
```

Comment:
I realise you are trying to help prevent others facing the same issue but this warning is invalid. It is ok to put/get configuration for an application that isn't loaded. This might be used when repeatedly setting persistent configuration before loading/starting applications. It is probably more appropriate to add a warning to exrm docs.

The issue you experienced was because configuration defined using `Mix.Config` (not `Application`) got "lost" when used by an external task. So I feel if any note was to be added to elixir, rather than the external task, it would be to `Mix.Config`. However I think it would require rephrasing to a guide on how a task might use the config rather than about quirks/features of external tools.